week-1-ts-ecosystem/w01e01-typescript-basics/src/userLogic.ts

```diff
@@ -0,0 +1,9 @@
+import {User} from "./type";
+
+export const hasAddress =(address: User["address"]): boolean => {
+    return !!address
+}
+
+export const hasGivenAge = (requiredAge: User["age"]): (userAge: User["age"]) => boolean => {
+    return (userAge: User["age"]): boolean => userAge >= requiredAge
+}
```

Comment:
`userLogic` można bardziej zawęzić w nazwie - jako `userValidators.ts`.
Nie wiem czy jest sens powtarzania zapisu typu tej funkcji:
```
: (userAge: User["age"]) => boolean
```
Dlatego że to wynika z tego co piszesz w `properties` i jest wyraźnie widoczne.
Natomiast pamiętaj że `typescript` ma tzw. wnioskowanie typów.

Więc nie musisz mu wszędzie pisać jawnie typu.
Tutaj akurat bym się powtrzymał bo obniża to czytelność.